### PR TITLE
Finished Assessment

### PR DESCRIPTION
**Pull request by nataliadiaz2**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,35 +15,98 @@
...
 // a) Create a test with an expect statement using the variable provided. HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
-
+describe("shuffledArray", () => {
+ it("takes in an array, removes the first item from the array and shuffles the remaining content", () => {
+   const colors1 = ["purple", "blue", "green", "yellow", "pink"]
```
Comment (SunkissedQueen) [inline code-challenges.test.js:20]: Great work on the test structure. Appreciate that you placed the variables within the test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,35 +15,98 @@
...
+   // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+   const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+   // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+   expect(shuffledArray(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+   expect(shuffledArray(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+ })
+})
+
+// ● shuffledArray › takes in an array, removes the first item from the array and shuffles the remaining content
```
Comment (SunkissedQueen) [inline code-challenges.test.js:29]: Thanks for leaving your test suite results

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,35 +15,98 @@
...
 
+//Create a function - shuffledArray
+//Parameter - array
+//Use the .slice() method to take remove the first item from the array
+//Use the .sort() method followed by Math.random() to randomize the strings in the sliced array.
+
+const shuffledArray = (array) => {
+  let sliced = array.slice(1)
+  return sliced.sort(() => .5 - Math.random());
+}
```
Comment (SunkissedQueen) [inline code-challenges.test.js:43]: Did this pass? It looks like it should.

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,35 +15,98 @@
...
+  it("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    // Expected output: [-8, 90]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    // Expected output: [5, 109]
+    expect(minAndMax(nums1)).toEqual([-8, 90])
+    expect(minAndMax(nums2)).toEqual([5, 109])
+  })
+})
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:59]: You got that structure!

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,35 +15,98 @@
...
+// ● minAndMax › takes an array of numbers and returns an array of the minimum and maximum numbers in that order
+//
+//    ReferenceError: minAndMax is not defined
```
Comment (SunkissedQueen) [inline code-challenges.test.js:62]: Good failure!!

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,35 +15,98 @@
...
+//Sort the min and max numbers
+//Return the new array with just the two filtered numbers
+
+const minAndMax = (array) => {
+  let total = []
+  let min = total.push(Math.min.apply(Math, array))
+  let max = total.push(Math.max.apply(Math, array))
+  return total.sort(function (a,b){return a-b})
+}
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:80]: Just for future notes, I recommend you including an explanation of why you had to also use .apply with Math.min and Math.max

Comment (SunkissedQueen) [inline code-challenges.test.js:80]: Did you get a green pass?

File: code-challenges.test.js [hunk 7]
```diff
@@ -15,35 +15,98 @@
...
 
 // b) Create the function that makes the test pass.
+
+//Create a function - combinedArray
+//Parameter - Array
+//Combine both arrays
+//Remove duplicate values with a spread operator
+//Return one array
+
+const combinedArray = (array1, array2) => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:109]: Wonderful work. Did it pass?

File: interview-questions.md [hunk 8]
```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props is a way to reference a class in a separate component.
 
-  Researched answer:
+  Researched answer: Props in React is a process of passing data/information from a component to another component, only able to move from parent to child class in one direction. Props are not able to be changed/updated through the child class, but only through the parent class.
```
Comment (SunkissedQueen) [inline interview-questions.md:12]: Great explanation

File: ruby-challenges.rb [hunk 9]
```diff
@@ -10,6 +10,17 @@
 num3 = 221
 # Expected output: '221 is odd'
 
+def num_determine number
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:13]: great structure. Woohoo you remembered to print the function call with an argument

File: ruby-challenges.rb [hunk 10]
```diff
@@ -21,7 +32,13 @@
 album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
 
+def vowel_remover string
+  string.delete "aeiouAEIOU"
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:36]: yes covered all cases

File: ruby-challenges.rb [hunk 11]
```diff
@@ -31,3 +48,15 @@
...
 # Expected output: 'Rotator is a palindrome'
+
+def possible_palindrome string
+  if string.downcase == string.downcase.reverse
+    "#{string} is a palindrome"
+  else
+    "#{string} is not a palindrome"
+  end
+end
+
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:59]: quality work

**Review by SunkissedQueen — APPROVED** [review]
Natalia, I like the way you approached this assessment. Your logic is understandable. your pseudo-code is easy to follow. You shared your knowledge in the interview questions. Solid work. Now we will work on sharing more of your voice on the interview by either sharing your experience or fun facts that you learned during research. Great job!

File: code-challenges.test.js (hunk 3 above)
Comment (nataliadiaz2) [reply code-challenges.test.js:43]: It did!

**Pull request merged**